bin/proving-service/src/proxy/status.rs

```diff
@@ -1,103 +1,226 @@
+///
+/// This is responsible for periodically updating the status of the proxy.
+pub struct ProxyStatusUpdater {
+    /// The load balancer state.
+    ///
+    /// This is used to generate the status response.
+    load_balancer: Arc<LoadBalancerState>,
+    /// The status transmitter.
+    ///
+    /// This is used to send the status updates to the proxy status service.
+    status_tx: watch::Sender<ProxyStatusResponse>,
+    /// The interval at which to update the status.
+    update_interval: Duration,
+}
+
+impl ProxyStatusUpdater {
+    /// Creates a new [`ProxyStatusUpdater`].
+    pub fn new(
+        load_balancer: Arc<LoadBalancerState>,
+        status_tx: watch::Sender<ProxyStatusResponse>,
+    ) -> Self {
+        Self {
+            load_balancer,
+            status_tx,
+            update_interval: Duration::from_secs(STATUS_UPDATE_INTERVAL_SECS),
         }
+    }
+
+    /// Starts the status updater.
+    ///
+    /// This is responsible for periodically updating the status of the proxy.
+    pub async fn start(&self, mut shutdown: watch::Receiver<bool>) {
+        let mut update_timer = interval(self.update_interval);
+        loop {
+            tokio::select! {
+                _ = update_timer.tick() => {
+                    let new_status = generate_status(&self.load_balancer).await;
+                    let _ = self.status_tx.send(new_status);
+                }
+                _ = shutdown.changed() => {
+                    info!("Status updater received shutdown signal");
+                    break;
+                }
+            }
+        }
+    }
+}
 
-        if session.req_header().uri.path() != "/status" {
-            let _ = session.respond_error(404).await;
-            return None;
+// UTILS
+// ================================================================================================
+
+impl From<&RustWorkerHealthStatus> for WorkerHealthStatus {
+    fn from(status: &RustWorkerHealthStatus) -> Self {
+        match status {
+            RustWorkerHealthStatus::Healthy => WorkerHealthStatus::Healthy,
+            RustWorkerHealthStatus::Unhealthy { .. } => WorkerHealthStatus::Unhealthy,
+            RustWorkerHealthStatus::Unknown => WorkerHealthStatus::Unknown,
         }
+    }
+}
+
+/// Generates a new status from the load balancer and returns it as a [`ProxyStatusResponse`].
+async fn generate_status(load_balancer: &LoadBalancerState) -> ProxyStatusResponse {
+    let workers = load_balancer.workers.read().await;
+    let worker_statuses: Vec<WorkerStatus> = workers.iter().map(WorkerStatus::from).collect();
+
+    let supported_proof_type: ProofType = load_balancer.supported_prover_type.into();
 
-        let _ = self.handle_request(&mut session).await.map_err(|e| {
-            error!("Failed to handle status request: {}", e);
-            None::<Stream>
-        });
-        None
+    ProxyStatusResponse {
+        version: env!("CARGO_PKG_VERSION").to_string(),
```

Comment:
nit: Could access the env var just once in `ProxyStatusUpdater::new()` or even as an argument provided to that fn.